compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -999,9 +999,9 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
        * FIXME This could clash with a local variable of the constructor or a JS
        * class capture. How do we avoid this?
        */
-      val selfName = freshLocalIdent("this")(pos)
+      val selfIdent = freshLocalIdent("this")(pos)
       def selfRef(implicit pos: ir.Position) =
-        js.VarRef(selfName)(jstpe.AnyType)
+        js.VarRef(selfIdent.name)(jstpe.AnyType)
```

Comment:
IIUC this will change the pos of the synthetic `this` from pos of the anon class creation to the pos of the definition using it (fields / props).

IMO this is acceptable (or even desirable).